### PR TITLE
view.pickFeaturesAt() -> avoid picking twice the same featureGeometry

### LINKED ISSUE #2093: PickFeatureAt: Duplication of Feature
A bug in view.pickFeatureAt() make it, in some special case, send twice the same featureGeometry.
That appear when when request for a point geometry which is situated close to the tile border as each tile is review separately and the  radius can make if find a point which isn't on this specific tile.

Example:
image

Here we can see that the plugin FeatureToolBox using the pickFeatureAt() show the duplication of the properties as if 2 distinct features got picked.

### PR DESCRIPTION
**Pull request by ftoromanoff**
link with the issue #2093

### CODE REVIEW

**Review by gchoqueux — COMMENTED** [review]
may be modify `FeaturesUtils.isFeatureUnderCoordinate` to avoid same feature adding?

**Comment by ftoromanoff** [comment]
> may be modify `FeaturesUtils.isFeatureUnderCoordinate` to avoid same feature adding?

At first I was thinking the same, but there is no probleme with it. The duplicatation comes from:
```for (const tile of tiles)``` at View.js line 914.
Or if we went this it would be necessary to add the tile extent to truncate the reserch circle at the tile extent, but i tought it would be more complicated..

**Comment by gchoqueux** [comment]
may be the issue come from `pickTilesAt` that returns too much tiles?

File: src/Core/View.js [hunk 1]
```diff
@@ -920,8 +926,12 @@ class View extends THREE.EventDispatcher {
 
                     precision = CRS.isMetricUnit(texture.features.crs) ? precisions.M : precisions.D;
 
-                    result[materialLayer.id] = result[materialLayer.id].concat(
-                        FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision));
+                    const featuresUnderCoor = FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision);
+                    featuresUnderCoor.forEach((feature) => {
+                        if (!result[materialLayer.id].map(f => f.geometry).includes(feature.geometry)) {
```
Comment (gchoqueux) [inline src/Core/View.js]: could you use `Array.find`?

Comment (ftoromanoff) [reply src/Core/View.js]: Done

**Review by gchoqueux — APPROVED** [review]

File: src/Core/View.js [hunk 2]
```diff
@@ -920,8 +926,12 @@ class View extends THREE.EventDispatcher {
 
                     precision = CRS.isMetricUnit(texture.features.crs) ? precisions.M : precisions.D;
 
-                    result[materialLayer.id] = result[materialLayer.id].concat(
-                        FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision));
+                    const featuresUnderCoor = FeaturesUtils.filterFeaturesUnderCoordinate(coordinates, texture.features, precision);
+                    featuresUnderCoor.forEach((feature) => {
+                        if (!result[materialLayer.id].find(f => f.geometry === feature.geometry)) {
```
Comment (jailln) [inline src/Core/View.js:935]: Shouldn't we add an id to `Feature` (that  would be generated) to simplify such comparisons ? (can be done in another PR)

File: src/Core/View.js [hunk 3]
```diff
@@ -832,8 +832,9 @@ class View extends THREE.EventDispatcher {
     }
 
     /**
-     * Searches for {@link Feature} in {@link ColorLayer}, under the mouse of at
-     * a specified coordinates, in this view.
+     * Searches for {@link FeatureGeometry} in {@link ColorLayer}, under the mouse of at
+     * a specified coordinates, in this view. Combining them per layer and in a Feature
+     * like format.
```
Comment (jailln) [inline src/Core/View.js:841]: small typos: 
`of` -> `or`
`a specified coordinates` -> `a specified coordinate`

Comment (ftoromanoff) [reply src/Core/View.js:841]: corrected

File: src/Core/View.js (hunk 2 above)
Comment (ftoromanoff) [reply src/Core/View.js:935]: I thought about it as well, but not sure it is will easy to set up ans usefull for anything else.

**Review by jailln — APPROVED** [review]

**Pull request merged**